sdk-commons/src/main/java/org/openmrs/maven/plugins/utility/DistroHelper.java

```diff
@@ -299,213 +303,152 @@ public void saveDistroPropertiesTo(File destination, String distro) throws MojoE
 	}
 
 	/**
-	 * Parses and processes content properties from content packages defined in the given {@code DistroProperties} object.
-	 *
-	 * <p>This method creates a temporary directory to download and process content package ZIP files specified
-	 * in the {@code distroProperties} file. The method delegates the download and processing of content packages
-	 * to the {@code downloadContentPackages} method, ensuring that the content packages are correctly handled
-	 * and validated.</p>
-	 *
-	 * <p>After processing, the temporary directory used for storing the downloaded ZIP files is deleted,
-	 * even if an error occurs during processing. If the temporary directory cannot be deleted, a warning is logged.</p>
-	 *
-	 * @param distroProperties The {@code DistroProperties} object containing key-value pairs specifying
-	 *                         content packages and other properties needed to build a distribution.
-	 *
-	 * @throws MojoExecutionException If there is an error during the processing of content packages,
-	 *                                such as issues with creating the temporary directory, downloading
-	 *                                the content packages, or IO errors during file operations.
+	 * This validates the distribution based on distro properties, and indicates any incompatibilities in the declared version
+	 * Currently, this only validates content packages, though this could be expanded to validate modules based on config.xml, etc.
 	 */
-	public void parseContentProperties(DistroProperties distroProperties) throws MojoExecutionException {
-		File tempDirectory = null;
-		try {
-			tempDirectory = Files.createTempDirectory("content-packages").toFile();
-			downloadContentPackages(tempDirectory, distroProperties);
-
-		} catch (IOException e) {
-			throw new MojoExecutionException("Failed to process content packages", e);
-		} finally {
-			if (tempDirectory != null && tempDirectory.exists()) {
-				try {
-					FileUtils.deleteDirectory(tempDirectory);
-				} catch (IOException e) {
-					log.warn("Failed to delete temporary directory: {}", tempDirectory.getAbsolutePath(), e);
```

Comment:
This is a convenience method that gets the missing dependencies, iterates over them, formats these into friendly error messages for display, and throws an exception with this.